### PR TITLE
Add option to `list` CLI to filter by status

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
## Description

This PR allows users to more easily filter which optimisations currently have what status when using the `list` executor command.

## Status
- [X] Ready to go

### CODE REVIEW

File: openff/bespokefit/cli/executor/list.py [hunk 1]
```diff
@@ -49,11 +61,19 @@ def list_cli():
 
     settings = current_settings()
 
+    status_url = (
+        None
+        if status_filter is None
+        else status_filter.replace("success", "complete").replace("errored", "complete")
```
Comment (jthorton) [inline openff/bespokefit/cli/executor/list.py:67]: Just to make sure I understand, we keep errored tasks in the complete queue so we have to query that queue and then filter if the actual status of the task is errored?

Comment (jthorton) [reply openff/bespokefit/cli/executor/list.py:67]: slightly picky can I add a comment saying this as I am sure I will forget in future!

**Review by jthorton — APPROVED** [review]
Awesome change which will help find errors even faster!

File: openff/bespokefit/cli/executor/list.py (hunk 1 above)
Comment (SimonBoothroyd) [reply openff/bespokefit/cli/executor/list.py:71]: Absolutey! I've added one in fabcedc

**Pull request merged**